TLM/TMPE.API/Util/Functional.cs

```diff
@@ -0,0 +1,53 @@
+namespace TrafficManager.Util {
+    /// <summary>
+    /// Represents a void return type which may also carry an error.
+    /// </summary>
+    /// <typeparam name="VALUE">Value type on success</typeparam>
+    /// <typeparam name="ERROR">Error type if failed</typeparam>
+    public class VoidResult<ERROR> {
+        private bool isOk_;
```

Comment:
Which naming convention is right? 
Here you have underscore suffix but in Options class you use prefix 
I there any logic behind that?😃

Reply:
If Resharper complains about naming i often ignore.
If StyleCop complains about naming, i usually satisfy and do what it wants.
So mainly StyleCop rules apply.

Reply:
Answer to this also needs to go in contribution guide.

Reply:
General rules which i keep in mind (and they mostly match with StyleCop):
Const goes ALL_CAPS
Public goes AllCamelCase
Private goes either lowCamelCase or lowCamelCase_
StyleCop seems to insist sometimes that _ goes before like _lowCamelCase, there are so many small conditions on that i just follow the warnings in my IDE

Reply:
[By CamelCase I assume you mean PascalCase](https://medium.com/better-programming/string-case-styles-camel-pascal-snake-and-kebab-case-981407998841)?

Reply:
BTW Rider suggests `TValue` and `TError` for type param names

Reply:
Might just change that. I'm delaying use of this module until it becomes clear where it will fit. I want to use it in functions which return a `bool` and an `out ErrorType reason`, for example.